### PR TITLE
[C10D] clean up pointless 'or None' clause

### PR DESCRIPTION
**Pull request by wconstab**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #129522

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @yf225 @chauhang @d4l3k @nvcastet

### CODE REVIEW

**Review by awgu — APPROVED** [review]

**Comment by clee2000** [comment]
@pytorchbot merge

Cancelled old merge job to fix something, reissuing merge command, sorry about that

File: torch/distributed/distributed_c10d.py [hunk 1]
```diff
@@ -4394,7 +4394,7 @@ def _new_group_with_tag(
     global _world
 
     default_pg = _get_default_group()
-    device_id = default_pg.bound_device_id or None
+    device_id = default_pg.bound_device_id
```
Comment (yifuwang) [inline torch/distributed/distributed_c10d.py:4397]: Interesting.. was this setting `device_id` to `None` when `bound_device_id` is `0`?

**Pull request closed without merging**